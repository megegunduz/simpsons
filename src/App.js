import React, { useEffect } from 'react';
import { MainNavigation } from './Navigation';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { store, persistor } from './Redux/Store';
import { LoadingOverlay } from './Features/Loading';
import SplashScreen from 'react-native-splash-screen';

const App = props => {
    useEffect(() => {
        SplashScreen.hide();
    }, []);

    return (
        <Provider store={store}>
            <PersistGate persistor={persistor}>
                <LoadingOverlay />
                <MainNavigation />
            </PersistGate>
        </Provider>
    );
};

export default App;
