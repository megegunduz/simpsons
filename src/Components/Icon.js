import React from 'react';
import { TouchableOpacity } from 'react-native';

/**
 * @param {object} props
 * @param {any} props.svg
 * @param {{width: number, height: number, color: import('react-native').ColorValue}} [props.style]
 * @param {import('react-native').ViewStyle} [props.containerStyle]
 * @param {number} [props.width]
 * @param {number} [props.height]
 * @param {() => void} [props.onPress]
 */
const Icon = props => {
    const { svg: Svg, style, containerStyle, width, height, onPress } = props;

    return (
        <TouchableOpacity
            style={containerStyle}
            onPress={onPress}
            disabled={!onPress}
            activeOpacity={0.4}
            testID={'icon-touchable'}
        >
            <Svg style={style} width={width} height={height} />
        </TouchableOpacity>
    );
};

export default Icon;
