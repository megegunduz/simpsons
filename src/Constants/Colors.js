const ConstantColors = {
    white: '#FFFFFF',
    black: '#000000',
    gray: '#8A9197',
    green: '#46732E',
    red: '#C80813',
    yellow: '#FED439',
    brown: '#91331F',
    lightBrown: '#D2AF81',
    blue: '#709AE1',
    darkBlue: '#197EC0',
    lightBlue: '#71D0F5',
};

/**
 * @param {string} hexCode
 * @param {number} opacity - Opacity between 0 - 100;
 * @returns {string} rgba string
 */
export const toRGBA = (hexCode, opacity) => {
    let hex = hexCode.replace('#', '');

    if (hex.length === 3) {
        hex = `${hex[0]}${hex[0]}${hex[1]}${hex[1]}${hex[2]}${hex[2]}`;
    }

    const r = parseInt(hex.substring(0, 2), 16);
    const g = parseInt(hex.substring(2, 4), 16);
    const b = parseInt(hex.substring(4, 6), 16);

    return `rgba(${r},${g},${b},${opacity / 100})`;
};

export const Colors = {
    constantColors: {
        ...ConstantColors,
    },
};

export default {
    ...ConstantColors,
    toRGBA,
};
