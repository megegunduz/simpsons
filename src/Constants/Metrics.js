import { Dimensions, Platform, StatusBar } from 'react-native';

const { width, height } = Dimensions.get('window');
const statusBarHeight = StatusBar.currentHeight;
const designWidth = 360;

const measure = designSize => designSize * (width / designWidth);
const fullRadius = Math.round((width + height) / 2);

export default {
    measure,
    width,
    height,
    statusBarHeight,
    fullRadius,
    baseMargin: measure(10),
    semiMargin: measure(15),
    section: measure(50),
    border: measure(1),
    elevation: e => measure(Platform.Version >= 28 ? e : e / 3),
    headerHeight: measure(50),
};
