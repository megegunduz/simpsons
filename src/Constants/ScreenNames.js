export default {
    HOME: 'home-screen',
    DETAIL: 'detail-screen',
    ADD: 'add-screen',
    INITIAL_LOADING: 'initial-loading',
};
