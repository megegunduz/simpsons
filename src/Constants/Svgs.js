import Down from '../Assets/Svgs/Down';
import Plus from '../Assets/Svgs/Plus';
import Trash from '../Assets/Svgs/Trash';
import Up from '../Assets/Svgs/Up';
import Back from '../Assets/Svgs/Back';

export default { Down, Plus, Trash, Up, Back };
