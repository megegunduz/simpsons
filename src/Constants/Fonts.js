import Metrics from './Metrics';

const step = Metrics.width * 0.0025;
const zero = Metrics.width * 0.002;

const size = punto => zero + step * punto;

export default {
    size,
};
