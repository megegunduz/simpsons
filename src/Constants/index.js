import ScreenNames from './ScreenNames';
import Metrics from './Metrics';
import Colors from './Colors';
import Fonts from './Fonts';
import Svgs from './Svgs';

export { ScreenNames, Metrics, Colors, Fonts, Svgs };
