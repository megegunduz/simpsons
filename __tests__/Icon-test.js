import React from 'react';
import { Icon } from '../src/Components';
import { Svgs } from '../src/Constants';
import renderer from 'react-test-renderer';
import { fireEvent, waitFor, render } from '@testing-library/react-native';

describe('Icon', () => {
    it('renders correctly', () => {
        expect(
            renderer.create(<Icon svg={Svgs.Plus} onPress={() => {}} />),
        ).toMatchSnapshot();
    });

    it('onPress functionality', async () => {
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Pressed');
        const { getByTestId } = render(
            <Icon svg={Svgs.Trash} onPress={onPressEvent} />,
        );
        await waitFor(() => {
            expect(getByTestId('icon-touchable')).toBeTruthy();
        });
        fireEvent.press(getByTestId('icon-touchable'));
        expect(onPressEvent).toBeCalled();
    });
});

describe('Icon disabled', () => {
    it('should be disabled when onPress prop is not passed', () => {
        expect(renderer.create(<Icon svg={Svgs.Up} />)).toMatchSnapshot();
    });
});
