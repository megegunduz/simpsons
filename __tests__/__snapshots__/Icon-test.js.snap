// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon disabled should be disabled when onPress prop is not passed 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="icon-touchable"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={50}
    bbWidth={50}
    focusable={false}
    height={50}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 50,
          "width": 50,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={50}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup>
      <RNSVGPath
        d="M12 16q.425 0 .713-.288Q13 15.425 13 15v-3.2l.9.9q.275.275.7.275.425 0 .7-.275.275-.275.275-.7 0-.425-.275-.7l-2.6-2.6q-.15-.15-.325-.213-.175-.062-.375-.062t-.375.062q-.175.063-.325.213l-2.6 2.6q-.275.275-.275.7 0 .425.275.7.275.275.7.275.425 0 .7-.275l.9-.9V15q0 .425.288.712.287.288.712.288Zm0 6q-2.075 0-3.9-.788-1.825-.787-3.175-2.137-1.35-1.35-2.137-3.175Q2 14.075 2 12t.788-3.9q.787-1.825 2.137-3.175 1.35-1.35 3.175-2.138Q9.925 2 12 2t3.9.787q1.825.788 3.175 2.138 1.35 1.35 2.137 3.175Q22 9.925 22 12t-.788 3.9q-.787 1.825-2.137 3.175-1.35 1.35-3.175 2.137Q14.075 22 12 22Zm0-2q3.325 0 5.663-2.337Q20 15.325 20 12t-2.337-5.663Q15.325 4 12 4T6.338 6.337Q4 8.675 4 12t2.338 5.663Q8.675 20 12 20Zm0-8Z"
        fill={4282861383}
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Icon renders correctly 1`] = `
<View
  accessibilityState={
    {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="icon-touchable"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={50}
    bbWidth={50}
    focusable={false}
    height={50}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 50,
          "width": 50,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={50}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup>
      <RNSVGPath
        d="M12 19q-.425 0-.712-.288Q11 18.425 11 18v-5H6q-.425 0-.713-.288Q5 12.425 5 12t.287-.713Q5.575 11 6 11h5V6q0-.425.288-.713Q11.575 5 12 5t.713.287Q13 5.575 13 6v5h5q.425 0 .712.287.288.288.288.713t-.288.712Q18.425 13 18 13h-5v5q0 .425-.287.712Q12.425 19 12 19Z"
        fill={4282861383}
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
